app/services/trainees/create_from_hesa.rb

```diff
@@ -106,8 +117,21 @@ def personal_details_attributes
       }
     end
 
+    def lead_partner_attributes
+      return {} unless LEAD_PARTNER_TO_ACCREDITED_PROVIDER_MAPPING.keys.include?(hesa_trainee[:ukprn])
+
+      lead_partner = LeadPartner.find_by(urn: LEAD_PARTNER_TO_ACCREDITED_PROVIDER_MAPPING[hesa_trainee[:ukprn]][:urn])
+
+      lead_partner ? { lead_partner: } : {}
+    end
+
     def provider_attributes
-      provider = Provider.find_by(ukprn: hesa_trainee[:ukprn])
+      provider = if LEAD_PARTNER_TO_ACCREDITED_PROVIDER_MAPPING.keys.include?(hesa_trainee[:ukprn])
```

Comment:
Same as previous comment
```suggestion
      provider = if hesa_trainee[:ukprn].in?(LEAD_PARTNER_TO_ACCREDITED_PROVIDER_MAPPING)
```